src/expr/src/vector_op/agg/aggregator.rs

```diff
@@ -49,9 +49,10 @@ pub trait Aggregator: Send + 'static {
     fn update_and_output_with_sorted_groups(
         &mut self,
         input: &DataChunk,
+        offset: usize,
         builder: &mut ArrayBuilderImpl,
         groups: &EqGroups,
-    ) -> Result<()>;
+    ) -> Result<usize>;
```

Comment:
What's the returned value for?

Reply:
A row offset to the child chunk (the column to aggregate). Because we want to limit the size of an output chunk, we may stop building the agg state in the middle.

Reply:
This may be avoided:
https://github.com/singularity-data/risingwave/pull/2084#discussion_r860838029